### DISCUSSION TITLE
Can you assign different colors when you display a different column? This is convenient to view

### DISCUSSION THREAD

**Discussion by joytianya** [saulpw/visidata#1205] [Q&A]
Can you assign different colors when you display a different column? This is convenient to view

**Comment by joytianya** [comment]
Is this feature already? How to open when using the VD command?

**Comment by saulpw** [comment]
Hi @joytianya, I'm not sure I understand exactly what you what.  You might be looking for `--color-current-col`, which colors the cursor column differently than the rest of the columns; or you might be interested in the [Colors API](https://www.visidata.org/docs/api/interface#colors) to add "colorizers" which can color whole columns depending on the given lambda.  For instance you could put this in your .visidatarc:

```
Column.init('color')
Sheet.colorizers += [ColumnColorizer(5, None, lambda s,c,r,v: c and c.color)]
ColumnsSheet.columns += [AttrColumn('color')]
```

And then on the Columns Sheet (Shift+C), there's a new column "color", which you can use to color the columns on the source sheet however you like. Try `green` or `black on 63`.

**Reply by joytianya** [reply]
Thank you for your reply, I want to use the vd command to view the CSV file, each column is displaying different colors, so it is easier to distinguish different columns, how to do this?

**Reply by joytianya** [reply]
In addition, when viewing the CSV file, when a content is long, use the V key to automatically wrap and view the full content, but for Chinese content is invalid， what is the cause？

**Reply by saulpw** [reply]
Text wrap uses Python `textwrap` stdlib, so if that doesn't know how to wrap Chinese content correctly, then that is the problem.

**Reply by joytianya** [reply]
I want to use the vd command to view the CSV file, each column is displaying different colors, so it is easier to distinguish different columns, like vscode extension "rainbow csv", how to do this?

**Comment by geekscrapy** [comment]
Try this out by adding to your `.visidatarc` file:

`Sheet.addCommand(None, 'rainbow-cols', 'sheet.addColorizer(ColumnColorizer(100, None, lambda s,c,r,v: str(s.columns.index(c)+1) if s and c else None))', 'add colorizer that colors based on the column index')`

Use the command '`rainbow-cols`' on any sheet that you want coloured.

@saulpw, I was trying to auto apply this colorizer to just `CsvSheet`, but it doesn't seem to be exposed via `@VisiData.api` for ease of use. Is this intentional? If not I can take a look at making a PR for that and a few of the other Sheets if you want?

**Reply by geekscrapy** [reply]
![image](https://user-images.githubusercontent.com/11225502/144590756-0b3af912-4abe-49e8-a148-30ee86684d2c.png)

**Reply by geekscrapy** [reply]
This works for more than 15 columns and rotates through the colours.

`Sheet.addCommand(None, 'rainbow-cols', 'sheet.addColorizer(ColumnColorizer(100, None, lambda s,c,r,v: str(abs((int(s.columns.index(c)/15)*15)-s.columns.index(c))+1) if s and c else None))', 'add colorizer that colors based on the column index')`

**Reply by saulpw** [reply]
> @saulpw, I was trying to auto apply this colorizer to just CsvSheet, but it doesn't seem to be exposed via @VisiData.api for ease of use. Is this intentional? If not I can take a look at making a PR for that and a few of the other Sheets if you want?

Not sure what you mean by this...did you use `CsvSheet.colorizers += [ ... ]` ?

**Reply by geekscrapy** [reply]
Yea, you get the following exception:
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/visidata/settings.py", line 335, in loadConfigFile
    exec(code, _globals)
  File "/Users/geekscrapy/.visidatarc", line 3, in <module>
    CsvSheet.colorizers
NameError: name 'CsvSheet' is not defined
```
Also tried `vd.CsvSheet`, still no dice. But I can access `JsonSheet.colorizers` just fine, as it is added to vd globals at:
https://github.com/saulpw/visidata/blob/c9f76dce63e3dfa6ca98b9ac1ece002693700cd3/visidata/loaders/json.py#L151-L154

Where-as CsvSheet has no decorator or addGlobals:
https://github.com/saulpw/visidata/blob/c9f76dce63e3dfa6ca98b9ac1ece002693700cd3/visidata/loaders/csv.py#L24-L26

**Reply by saulpw** [reply]
Ah gotcha!  Sorry, hitting things in reverse order.  I wouldn't mind exposing all of the sheet types, but I don't want to commit to the internal implementation in the more esoteric loaders.  So I'm not sure.
